framework/Yii.php

```diff
@@ -5,8 +5,6 @@
  * @license http://www.yiiframework.com/license/
  */
 
-require __DIR__ . '/BaseYii.php';
```

Comment:
Why?

Reply:
It looks like we include autoload in [web application](https://github.com/yiisoft/yii2-app-basic/blob/master/web/index.php) and [console application](https://github.com/yiisoft/yii2/blob/master/framework/yii) before including this file. I think that `require` here not required.

Reply:
Yes but since `Yii` and `BaseYii` files aren't in any namespace, autoload won't handle them. It's done in 2.1 already and I don't think it's a good idea to touch it in 2.0.

Reply:
Please revert this one.

Reply:
Sorry, but I see that `BaseYii` is in `yii` namespace (https://github.com/yiisoft/yii2/blob/master/framework/BaseYii.php). Looks like I missed something. Would you be so kind to explain this  little bit more?

Reply:
You're actually right :) Just tried it. Thanks for being persistent.

Reply:
This is incorrect IMO. Autoloader is registered bellow, so you can't rely on it at this point - it works only because you have composer's autoloader which takes care about loading `BaseYii` class. So now you need to another autoloader to init Yii autoloader...

Reply:
@rob006 , I see definition of `psr-4` autoloader for `yii` namespace in composer.json [here](https://github.com/yiisoft/yii2/blob/master/framework/composer.json#L79) and [here](https://github.com/yiisoft/yii2/blob/master/composer.json#L99). About another autoloader to init `Yii` autoloader -- from my point of view it is ok, because we should use as less code as possible to create functionality, which we need 😄 .

Reply:
Oh, right right. Yii 2.0 can work w/o Composer and some developers actually use that... should not be removed in 2.0 because of that...

Reply:
@samdark I see that application include autoloader in framework [console script](https://github.com/yiisoft/yii2/blob/master/framework/yii#L13) and in [web frontend script](https://github.com/yiisoft/yii2-app-basic/blob/master/web/index.php#L7). Are you speaking about users, who implement their own frontend and console scripts? In this case, of course, I must return  `require` back.

And one additional question: which branch are you have the latest code? I think that it is better to fix anything in latest code instead of porting fixes from old branches. For example in branch 2.1 `require` in [`Yii.php`](https://github.com/yiisoft/yii2/blob/2.1/framework/Yii.php#L8) and `require` for [`Yii.php`](https://github.com/yiisoft/yii2/blob/2.1/framework/yii#L27) should be removed, because `Yii.php` loaded via [composer autoload](https://github.com/yiisoft/yii2/blob/2.1/framework/composer.json#L82).

Thank you

Reply:
There are two up to date branches: master (2.0) and 2.1. See http://www.yiiframework.com/news/156/yii-2-1-and-yii-2-0-feature-freeze/

Reply:
Composer is included in zip distribution so I think that's OK to keep removal of these lines.